### ISSUE TITLE
Certain Syntax breaks mattermost chat.

### ISSUE THREAD

**Issue by hankinator** [mattermost/mattermost-server#10856]
**#### Summary**
Certain line(lines of text) completely break mattermost channels/chats. 

**#### Steps to reproduce**
1. Open mattermost chat
2. Post `Aaaa_Aaaaaaaa___Aaaaaaaaaaa_aaaaa = A._aaaa.aa_aaa * ( - Aaaa_Aaaaaaaa___Aaaa_aaaaaa_AAa * A.Aaaa_Aaaaaaaa.aaaa__Aaaaaaaa__AAa__Aaaaaaaaaaa * Aaaa_Aaaaaaaa___AAa_aa * Aaaa_Aaaaaaaa___AAa / Aaaa_Aaaaaaaa___AAAaaaaaaaaaAaaaa ) + A._aaaa.aa_aaa * ( - Aaaa_Aaaaaaaa___Aaaa_aaaaaa_Aaaaaaaaaaaaaa * A.Aaaa_Aaaaaaaa.aaaa__Aaaaaaaa__Aaaaaaaaaaaaaa__Aaaaaaaaaaa * Aaaa_Aaaaaaaa___Aaaaaaaaaaaaaa * Aaaa_Aaaaaaaa___Aaaaaaaaaaa / ( A.Aaaa_Aaaaaaaa.AA__Aaaaaaaa__Aaaaaaaaaaaaaa__Aaaaaaaaaaa + Aaaa_Aaaaaaaa___Aaaaaaaaaaa ) ) + A._aaaa.aa_aaa * ( - Aaaa_Aaaaaaaa___aaaaa__Aaaaaaaaaaa * Aaaa_Aaaaaaaa___Aaaaaaaaaaa ) + A._aaaa.aa_aaaa * ( ( A.Aaaa_Aaaaaaaa.aa__Aaaaaaaaaaa / Aaaa_Aaaaaaaa___Aaaaaaaaaaa ) ** ( a ) * A.Aaaa_Aaaaaaaa.Aaaaaaaaaaa_aaaaaa * A.Aaaa_Aaaaaaaa.aaaaaa_aaaaaa * ( Aaaa_Aaaaaaaa___Aaaa_aaaaaa ** a.a ) ) + A._aaaa.aa_aaaa * ( A.Aaaa_Aaaaaaaa.aaa_aaaaa_Aaaaaaaaaaa * ( A.Aaaa_Aaaaaaaa.aa__Aaaaaaaaaaa - Aaaa_Aaaaaaaa___Aaaaaaaaaaa ) * A.Aaaa_Aaaaaaaa.aaaaaa_aaaaaa )

`
Without the ``` around it in Mattermost chat

Both Mattermost browser and the Mattermost app breaks. 

**#### Expected result*
Text is posted without any errors.

**#### Observed behavior** 
The app/browser hangs and crashes.

**#### Possible fixes**
Unknown

**#### Workarounds**
Abandon the channel

**#### Mattermost Versions**
Mattermost Version: 5.9.0
Database Schema Version: 5.9.0
Database: postgres

**#### Additional information**
Using the marked demo at https://marked.js.org/demo/ the text above renders fine (on version 0.6.1 through 0.3.4). 

It hangs on markdown version 0.3.3 and below to 0.1.9. 

Anything below 0.1.9 works fine.

**Comment by felibb** [comment]
Same on 5.8.0

**Comment by megari** [comment]
This bug is actually really nasty. It will basically ruin a channel permanently, unless the offending message is somehow deleted from the channel. The only other option seems to be to archive the channel from the CLI and start over.

**Comment by amyblais** [comment]
Hi all, what OS and version are you seeing this on?

**Comment by felibb** [comment]
5.8.0, macOS 10.12.6 desktop client tested.

**Comment by jdepro** [comment]
This is also an issue on v4.2.1, Windows 10 1809 Desktop client

**Comment by hankinator** [comment]
@amyblais Hey there,

We see this on both Windows and Linux (Ubuntu in our case)

For a client: 
**Chrome:** Version 73.0.3683.86 (Official Build) Built on Ubuntu , running on Ubuntu 18.04 (64-bit)
**Firefox:** 66.0.5 (64-bit)

Mattermost is running on a ubuntu 16.04 LXC container.

**Comment by amyblais** [comment]
I'm not able to reproduce this but I would like to open a bug ticket since so many people are seeing this.

Are there any Mattermost log errors or any additional reproduction steps you could help share that I could include in a bug ticket?

**Comment by felibb** [comment]
[https://gist.github.com/felibb/b2e6a1e16847f82727fdfb62ee1af329](Gist)

This is what I found in `/opt/mattermost/logs/mattermost.log` from the time we tested. I don't remember neither stopping nor starting the server manually though.

I had to archive the channel immediately via CLI, as my client was unusable. After that the log looks normal.

**Comment by megari** [comment]
Hmm, this looks a bit suspect in the log @felibb posted:
```
2019-05-16T15:31:39.033+0300    warn    app/post_metadata.go:85 Failed to get embedded content for a post       {"post_id": "5dgft1dnbty4by4rerb5kptcnc", "err": "Get https://[domain]/[team]/channels/bug-test: address forbidden, you may need to set AllowedUntrustedInternalConnections to allow an integration access to your internal network"}
```

**Comment by hankinator** [comment]
This is apparently fixed in the newest version (5.9.1, it also works with 5.9.0). It was breaking with previous versions. If you update mattermost to (5.9.1 and 5.9.0) it should work. 

The issue reproduces with 5.8.2 (tested with a docker container) 

My apologies for the false information on the first post (says 5.9.0). Mattermost was patched between when I was gathered the info and when the bug first occured.

**Comment by amyblais** [comment]
Hi @hankinator, no worries, awesome to hear that it's fixed! That also explains why I couldn't reproduce it. I'll close this issue for now,
